src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -267,7 +270,42 @@ public void cancel(Account account, OCFile file) {
                 }
             }
         }
-        
+
+        /**
+         * Cancels a pending or current upload for an account
+         *
+         * @param account Owncloud accountName where the remote file will be stored.
+         */
+        public void cancel(Account account) {
+            Log_OC.d(TAG, "Account= " + account.name);
+
+            if (mCurrentDownload != null) {
+                Log_OC.d(TAG, "Current Download Account= " + mCurrentDownload.getAccount().name);
+                if (mCurrentDownload.getAccount().name.equals(account.name)) {
+                    mCurrentDownload.cancel();
+                }
+            }
+            // Cancel pending downloads
+            ConcurrentMap downloadsAccount = mPendingDownloads.get(account);
+            Iterator<String> it = downloadsAccount.keySet().iterator();
+            Log_OC.d(TAG, "Number of pending downloads= " + downloadsAccount.size());
+            while (it.hasNext()) {
+                String key = it.next();
+                Log_OC.d(TAG, "download CANCELLED " + key);
+                if (key.startsWith(account.name)) {
+                    DownloadFileOperation download;
```

Comment:
Please, take this line out of the ```while``` loop; reference declaration is not completely "for free".